### DISCUSSION TITLE
Accessing systems file problem

### DISCUSSION THREAD

**Discussion by Aminmalek** [canonical/docker-snap#244] [General]
I submitted the issue here [Link](https://github.com/sirixdb/sirix/issues/632)
The app in container wants to create some files out of container and there is some problems with docker installed by snap.

**Comment by zhsj** [comment]
Confirmed, looks like same as https://github.com/docker-snap/docker-snap/issues/81

It's affected by https://github.com/snapcore/snapd/blob/bc82b4432397ca50a696b81669df8600a775a0e1/interfaces/apparmor/template.go#L704


```
root@lunar-vm-1:~# docker run --rm -it busybox 'mkdir /opt/abc'
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "mkdir /opt/abc": stat mkdir /opt/abc: no such file or directory: unknown.
ERRO[0000] error waiting for container: context canceled 
root@lunar-vm-1:~# 
```

**Comment by Aminmalek** [comment]
What can I do to fix this?

**Comment by xnox** [comment]
> What can I do to fix this?

currently you can choose to install docker unconfined with `--devmode` or use a .deb version of docker. As at the moment to resolve this we will require snapd changes to change our confinement.

**Comment by xnox** [comment]
Oh, i wonder if you can "just" adjust the snapd generated docker apparmor profiles and restart it..... but also this is very hacky.
